### PR TITLE
Reserve 'aperture-cloud' agent group for cloud agent

### PR DESCRIPTION
**Pull request by hdkshingala**
##### Checklist

- [x] Tested in playground or other setup

### Summary by CodeRabbit

- New Feature: Introduced a new option, `CLOUD_AGENT`, for the `heartbeat_interval` and `heartbeat_mode` parameters, providing more flexibility for different deployment scenarios.
- Update: Changed the default agent group for the peer-to-peer cluster from "default" to "aperture-cloud".
- Update: Added validation checks to prevent the use of reserved group names and ensure appropriate configuration settings.
- New Feature: Added a new field `Type` to the `DeploymentConfigSpec` struct, allowing for more precise deployment configurations.
- Update: Improved code readability by replacing string literals with constants in several places.
- Documentation: Updated various documentation files to reflect the new changes and options.
  
These changes enhance the software's flexibility and usability, particularly in different deployment scenarios. They also improve code readability and maintainability.

### CODE REVIEW

File: pkg/agent-info/agent-info.go [hunk 1]
```diff
@@ -31,11 +41,16 @@
...
+func ProvideAgentInfo(unmarshaller config.Unmarshaller, configOverride *InstallationModeConfig) (*AgentInfo, error) {
 	var config AgentInfoConfig
 	if err := unmarshaller.UnmarshalKey(configKey, &config); err != nil {
 		return nil, err
 	}
+
+	if configOverride != nil && configOverride.InstallationMode != utils.InstallationModeCloudAgent && config.AgentGroup == utils.ApertureCloudAgentGroup {
+		return nil, fmt.Errorf("'%s' is a reserved group name for FluxNinja Cloud Agents. Please use a different agent group name", utils.ApertureCloudAgentGroup)
+	}
+
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply pkg/agent-info/agent-info.go:53]: If the ConfigOverride is nil that means the extension is not enabled and we don't have to do this check in that case. So the current condition is correct

File: operator/controllers/mutatingwebhook/agent_pod.go [hunk 2]
```diff
@@ -144,6 +145,10 @@ func agentPod(instance *v1alpha1.Agent, pod *corev1.Pod) error {
 		agentGroup = pod.Annotations[controllers.AgentGroupKey]
 	}
 
+	if agentGroup == utils.ApertureCloudAgentGroup {
```
Comment (kwapik) [inline operator/controllers/mutatingwebhook/agent_pod.go:148]: Don't we need the installation mode check here?

File: pkg/utils/consts.go [hunk 3]
```diff
@@ -1,7 +1,20 @@
...
+	// ApertureAgent is the service name for Agent.
+	ApertureAgent = "aperture-agent"
+	// ApertureController is the service name for Controller.
 	ApertureController = "aperture-controller"
+
+	// ApertureCloudAgentGroup agent group name for Cloud Agents.
+	ApertureCloudAgentGroup = "aperture-cloud"
+
+	// InstallationModeKubernetesSidecar for sidecar installation mode.
+	InstallationModeKubernetesSidecar = "KUBERNETES_SIDECAR"
```
Comment (kwapik) [inline pkg/utils/consts.go:13]: :+1: This makes code much cleaner

**Review by kwapik — APPROVED** [review]

File: operator/controllers/mutatingwebhook/agent_pod.go (hunk 2 above)
Comment (kwapik) [reply operator/controllers/mutatingwebhook/agent_pod.go:148]: Okay! Just wanted to double check.

**Pull request merged**